tutorials/general/t8_step6_stencil.cxx

```diff
@@ -0,0 +1,424 @@
+            break;
+          case 1: // SOUTH
+            stencil[2][1] = height;
+            dx[2] = element_data[neighids[0]].dx;
+            break;
+          case 2: // WEST
+            stencil[1][0] = height;
+            dy[0] = element_data[neighids[0]].dy;
+            break;
+          case 3: // EAST
+            stencil[1][2] = height;
+            dy[2] = element_data[neighids[0]].dy;
+            break;
+        }
+
+        T8_FREE(neighbors);
+        T8_FREE(dual_faces);
+        T8_FREE(neighids);
+      }
+
+      const double xslope_m = 0.5/(dx[0] + dx[1])*(stencil[1][1] - stencil[0][1]);
+      const double xslope_p = 0.5/(dx[1] + dx[2])*(stencil[2][1] - stencil[1][1]);
+
+      const double yslope_m = 0.5/(dy[0] + dy[1])*(stencil[1][1] - stencil[1][0]);
+      const double yslope_p = 0.5/(dy[1] + dy[2])*(stencil[1][2] - stencil[1][1]);
+
+      const double xslope = 0.5*(xslope_m + xslope_p);
+      const double yslope = 0.5*(yslope_m + yslope_p);
+
+      /* TODO: Probably still not optimal at non-conforming interfaces. */
+      const double xcurve = (xslope_p - xslope_m)/0.25/(dx[0] + 2.0*dx[1] + dx[2]);
+      const double ycurve = (yslope_p - yslope_m)/0.25/(dy[0] + 2.0*dy[1] + dy[2]);
+
+      element_data[current_index].schlieren = sqrt(xslope*xslope + yslope*yslope);
+      element_data[current_index].curvature = sqrt(xcurve*xcurve + ycurve*ycurve);
+    }
+  }
+}
+
+/* Each process has computed the data entries for its local elements.
+ * In order to get the values for the ghost elements, we use t8_forest_ghost_exchange_data.
+ * Calling this function will fill all the ghost entries of our element data array with the
+ * value on the process that owns the corresponding element. */
+static void
+t8_step6_exchange_ghost_data (t8_forest_t forest, struct data_per_element *data)
+{
+  sc_array           *sc_array_wrapper;
+  t8_locidx_t         num_elements = t8_forest_get_local_num_elements (forest);
+  t8_locidx_t         num_ghosts = t8_forest_get_num_ghosts (forest);
+
+  /* t8_forest_ghost_exchange_data expects an sc_array (of length num_local_elements + num_ghosts).
+   * We wrap our data array to an sc_array. */
+  sc_array_wrapper = sc_array_new_data (data, sizeof (struct data_per_element), num_elements + num_ghosts);
+
+  /* Carry out the data exchange. The entries with indices > num_local_elements will get overwritten. */
+  t8_forest_ghost_exchange_data (forest, sc_array_wrapper);
+
+  /* Destroy the wrapper array. This will not free the data memory since we used sc_array_new_data. */
+  sc_array_destroy (sc_array_wrapper);
+}
+
+/* Write the forest as vtu and also write the element's volumes in the file.
+ * 
+ * t8code supports writing element based data to vtu as long as its stored
+ * as doubles. Each of the data fields to write has to be provided in its own
+ * array of length num_local_elements.
+ * We support two types: T8_VTK_SCALAR - One double per element
```

Comment:
1 double or Three doubles (I think the second one is better)

Reply:
Done.